File: shared/javascript/src/docModel.js

goog.provide("pebble.shared.DocModel");

goog.require("pebble.Pebble");
goog.require("pebble.shared.interfaces.IServerData");


//-----file start
pebble.shared.DocModel = {};

/**
 * @type {pebble.shared.interfaces.IServerData}
 */
pebble.shared.DocModel.deployment = null;

/**
 * Factory must set datasources and deployment datasource
 * @return {pebble.shared.interfaces.IServerData}
 */
pebble.shared.DocModel.getDatasource = function() {

	return pebble.shared.DocModel.deployment;
};
/**
 * @param {string} fullId
 * @return {pebble.Pebble}
 */
pebble.shared.DocModel.getDoc = function(fullId) {
	try{
		var lastDeliminatorIndex = fullId.lastIndexOf(".");

		var container = "";
		var uname = fullId;
		if (lastDeliminatorIndex > 0) {
			//
			container = fullId.substring(0, lastDeliminatorIndex);
			uname = fullId.substring(lastDeliminatorIndex + 1);
		} else {
			//appInstances
			container = "theApp.theControlApp.appInstances";
			uname = fullId;
		}

		var ds = pebble.shared.DocModel.getDatasource();
		return ds.retrieve(container, uname);
	} catch (e) {
		return null;
	}
};

/**
 * @param {pebble.shared.interfaces.IServerData} depl
 */
pebble.shared.DocModel.setDeployment = function(depl) {
	pebble.shared.DocModel.deployment = depl;
};

File: shared/csharp/src/DocModel.cs
using System.Collections.Generic;
using PebbleFields.Shared.Interfaces;
using PebbleFields.PebbleObject;

namespace PebbleFields.Shared
{
	/**
	 * Doc is the unit document saved to the data store.
	 * It is simple crud but manages the default variables (ownerId, etc.)
	 * 
	 * @author Ryan Tsunoda
	 *
	 */
	public class DocModel : Pebble 
	{

		private static IServerData deployment;

		//path for create, update, delete, query > collection path
		//path for retrieve should be done thru getDoc(fullId)
		//Factory must set datasources and deployment datasource
		public static IServerData getDatasource() 
		{
				return deployment;
		}

		public static Pebble getDoc(string fullId) 
		{
			try
			{
				int lastDeliminatorIndex = fullId.LastIndexOf(".");

				string container = "";
				string uname = fullId;
				if (lastDeliminatorIndex > 0) 
				{
					//
					container = fullId.Substring(0, lastDeliminatorIndex);
					uname = fullId.Substring(lastDeliminatorIndex + 1);
				} else {
					//appInstances
					container = "theApp.theControlApp.appInstances";
					uname = fullId;
				}

				return DocModel.getDatasource().retrieve(container, uname);
			} 
			catch (System.Exception e) 
			{
				return null;
			}
		}
		public static void setDeployment(IServerData depl) 
		{
			deployment = depl;
		}
	}
}
